exchange.c

```diff
@@ -103,29 +103,22 @@ int domain_exchange(int (*layoutfunc)(int p)) {
     return failure;
 }
 
+#define INIT_EXTRA(NAME) int * NAME =  (int *) ta_malloc("NAME", int, NTask * sizeof(int)); int * NAME ## _sph = (int *) ta_malloc("NAME ## _sph", int, NTask * sizeof(int));int * NAME ## _bh = (int *) ta_malloc("NAME ## _bh", int, NTask * sizeof(int))
+
 static int domain_exchange_once(int (*layoutfunc)(int p), int* toGo, int * toGoSph, int * toGoBh, int *toGet, int *toGetSph, int *toGetBh)
 {
-    int count_togo = 0, count_togo_sph = 0, count_togo_bh = 0, 
+    int count_togo = 0, count_togo_sph = 0, count_togo_bh = 0,
         count_get = 0, count_get_sph = 0, count_get_bh = 0;
 
     int i, target;
     struct particle_data *partBuf;
     struct sph_particle_data *sphBuf;
     struct bh_particle_data *bhBuf;
 
-    int * count = ta_malloc("var", int, NTask);
-    int * count_sph = ta_malloc("var", int, NTask);
-    int * count_bh = ta_malloc("var", int, NTask);
-    int * offset = ta_malloc("var", int, NTask);
-    int * offset_sph = ta_malloc("var", int, NTask);
-    int * offset_bh = ta_malloc("var", int, NTask);
-
-    int * count_recv = ta_malloc("var", int, NTask);
-    int * count_recv_sph = ta_malloc("var", int, NTask);
-    int * count_recv_bh = ta_malloc("var", int, NTask);
-    int * offset_recv = ta_malloc("var", int, NTask);
-    int * offset_recv_sph = ta_malloc("var", int, NTask);
-    int * offset_recv_bh = ta_malloc("var", int, NTask);
+    INIT_EXTRA(count);
+    INIT_EXTRA(offset);
+    INIT_EXTRA(count_recv);
```

Comment:
We really should have had one per particle species -- I was thinking of introducing an array and a loop somehow. But the macro, hinders thand...

Reply:
good point...